controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +75,39 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator contextConfigurator = new SSLContextConfigurator();
+                contextConfigurator.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                contextConfigurator.setKeyStorePass(loadPasswordFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(contextConfigurator, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
         } finally {
             LoggerHelpers.traceLeave(log, this.objectId, "startUp", traceId);
         }
     }
 
+    private String loadPasswordFromFile(String keyFilePasswordPath) {
+        // In case the path is not specified, return empty string. This means the password is not used.
+        // In case of error, return empty password. Which will fail the SSL connection if the password is expected.
+
+        if (Strings.isNullOrEmpty(keyFilePasswordPath)) {
+            return "";
+        }
+        File passwdFile = new File(keyFilePasswordPath);
+        if (passwdFile.length() == 0) {
+            return "";
+        }
+        try {
+            return new String(FileUtils.readFileToByteArray(passwdFile), Charsets.UTF_8).trim();
```

Comment:
What happens if the file is too large? This can easily lead to OOM exceptions and could possibly be used as a way to crash the process. Please have a check to validate that the file isn't too big. Figure out a reasonable maximum length (4KB?) and fail if the file is too big.

Reply:
Fixed. Added a check on the filesize. Currently it is 4MB.